python/cudf/cudf/core/column/categorical.py

```diff
@@ -1506,27 +1506,23 @@ def _concat(objs: MutableSequence[CategoricalColumn]) -> CategoricalColumn:
             offset=codes_col.offset,
         )
 
-    def _copy_type_metadata(
-        self: CategoricalColumn, other: ColumnBase
-    ) -> ColumnBase:
-        """Copies type metadata from self onto other, returning a new column.
-
-        In addition to the default behavior, if `other` is not a
-        CategoricalColumn, we assume other is a column of codes, and return a
-        CategoricalColumn composed of `other`  and the categories of `self`.
-        """
-        if not isinstance(other, cudf.core.column.CategoricalColumn):
-            other = column.build_categorical_column(
-                categories=self.categories,
-                codes=column.as_column(other.base_data, dtype=other.dtype),
-                mask=other.base_mask,
-                ordered=self.ordered,
-                size=other.size,
-                offset=other.offset,
-                null_count=other.null_count,
+    def _apply_type_metadata(
```

Comment:
@vyasr pointed out that this function name may imply that this operation is happening in-place, though it is not. Is there a better name for this function to communicate this?

Reply:
I like `as` or `with` for this, e.g. `_with_type_metadata`, translating to "the thing we get when we combine this column with this other metadata".

Reply:
+1 for some variant of `astype*` since that would also match with pandas and numpy. Perhaps `asdtype*` since that would be a bit more explicit.

Reply:
Think I'm going to go with `_with_type_metadata` as not to confuse with the actual `astype` method in `ColumnBase`.

Reply:
I like `_with_type_metadata`